### PR TITLE
docs: bundle static swagger-ui with docs [DET-4210]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

- ship keep a single copy of swagger-ui
- serve swagger-ui as part of docs
  - with a cloud/deploy specific config to disable "try it feature"
  - allow try it if /docs are viewed through an active cluster.

> docs.determined.ai/<version>/api will load up the static Swagger
<CLUSTER>/docs/api will load up the live version with which a user can interact

### TODO

- [x] maybe redirect /swagger-ui to /docs/swagger-ui

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by hamidzr** [comment]
@rb-determined-ai does circleci changes make sense to you? how would I test the deployed version? I ran make deploy in docs and inspected the generated docs and swagger-ui but haven't actually deployed it to any cloud

File: docs/Makefile [hunk 1]
```diff
@@ -19,6 +19,10 @@ build: build-examples build-helm
 	mkdir -p site/downloads/helm
 	cp ../helm/build/* site/downloads/helm
 	$(MAKE) sp-html
+	# pull in swagger-ui
+	# cp -r ../master/static/swagger-ui site/html
```
Comment (hamidzr) [inline docs/Makefile:23]: remove

File: .circleci/config.yml [hunk 2]
```diff
@@ -547,6 +546,7 @@ commands:
 
 jobs:
   build-docs:
+    # TODO this now depends on build-proto
```
Comment (hamidzr) [inline .circleci/config.yml:549]: there is no circleci feature to declare this dependency here is there?

File: docs/how-to/rest-apis.txt [hunk 3]
```diff
@@ -24,11 +24,13 @@ As Swagger puts it:
    Specification, with the visual documentation making it easy for back
    end implementation and client-side consumption."
 
-Reach the Swagger UI for your Determined installation by clicking the
-API icon from the Determined WebUI or by visiting the path `/swagger-ui`
-on your Determined cluster.
 
-.. image:: /assets/images/swagger-ui-location.png
+A static version of the Swagger UI is available bundled with this docs
```
Comment (hamidzr) [inline docs/how-to/rest-apis.txt:28]: @neilconway any suggestions here?

other feature-full version maybe?

File: docs/release-notes/1376-static-swagger-docs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+:orphan:
+
+**New Features**
+- Bundled a version of Swagger-ui to Determiend documentation.
```
Comment (hamidzr) [inline docs/release-notes/1376-static-swagger-docs:4]: maybe 
"Determined API documentation (Swagger-UI) is now bundled with the rest of docs and available at docs.determined.ai". ?

File: .circleci/config.yml (hunk 2 above)
Comment (rb-determined-ai) [reply .circleci/config.yml:549]: I'd drop all of the circleci changes and just let proto rebuild.  It's just a couple of seconds, right?

File: docs/Makefile [hunk 5]
```diff
@@ -19,6 +19,10 @@ build: build-examples build-helm
 	mkdir -p site/downloads/helm
```
Comment (rb-determined-ai) [inline docs/Makefile:19]: This `build` target needs to depend on a `build-proto` target.

Lol, while you are at it, please fix the `.PHONE: build-helm` to be `.PHONY: build-helm`.  Just saw that.

**Comment by hamidzr** [comment]
> I'd drop all of the circleci changes and just let proto rebuild. It's just a couple of seconds, right?

that'd make it simpler. the custom image we use to build `build-docs` doesn't have go lang in it though :thinking:

**Review by rb-determined-ai — COMMENTED** [review]
when I click on the api button from the webui, it redirects me to `http://localhost:8080/docs/swagger-ui` which shows a ton of errors in my browser console.

However, manually navigating to `http://localhost:8080/docs/swagger-ui/` (with a trailing slash) works fine.

Can we fix the non-trailing-slash version?

**Comment by hamidzr** [comment]
>  it redirects me to http://localhost:8080/docs/swagger-ui which shows a ton of errors in my browser console.

kind of an artifact of how echo does routing for static directories. I had to call the /docs/swagger-ui as a separate route to get this to work.

**Review by rb-determined-ai — APPROVED** [review]

File: docs/how-to/rest-apis.txt (hunk 3 above)
Comment (hamidzr) [reply docs/how-to/rest-apis.txt:28]: cc @vishnu2kmohan

File: docs/how-to/rest-apis.txt [hunk 6]
```diff
@@ -24,11 +24,13 @@ As Swagger puts it:
...
+`here
+<../swagger-ui/index.html>`__. If
+you have access to a running Determiend cluster you can checkout the
```
Comment (vishnu2kmohan) [inline docs/how-to/rest-apis.txt:31]:
```suggestion
you have access to a running Determined cluster you can checkout the
```

File: docs/how-to/rest-apis.txt (hunk 3 above)
Comment (vishnu2kmohan) [inline docs/how-to/rest-apis.txt:28]:
```suggestion
A static version of the Swagger UI is bundled with these docs
```

File: docs/how-to/rest-apis.txt [hunk 7]
```diff
@@ -24,11 +24,13 @@ As Swagger puts it:
...
+more feature-full by clicking the API icon from the Determined WebUI or
```
Comment (vishnu2kmohan) [inline docs/how-to/rest-apis.txt:32]:
```suggestion
live-interact version by clicking the API icon from the Determined WebUI or
```

**Pull request merged**